components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -185,21 +204,45 @@ public void registerWritePolicySelector(final WritePolicySelector writePolicySel
   @Override
   @Guarded(by = STARTED)
   public StorageTx openTx() {
-    return openStorageTx(null);
+    return openStorageTx(databaseInstanceProvider.get().acquire(), false);
   }
 
   @Override
   @Guarded(by = STARTED)
   public StorageTx openTx(final ODatabaseDocumentTx db) {
     checkNotNull(db);
-    return openStorageTx(db);
+    return openStorageTx(db, true);
   }
 
-  private StorageTx openStorageTx(final ODatabaseDocumentTx db) {
-    ODatabaseDocumentTx database = db;
-    if (database == null) {
-      database = databaseInstanceProvider.get().acquire();
+  /**
+   * Find content validator for repository format.
+   *
+   * If no format-specific validator is configured, the default is used.
+   */
+  @Nonnull
+  private ContentValidator validator() {
```

Comment:
One benefit of extracting this out as a component/service is that you could then pass that into StorageFacetImpl rather than passing the map of validators plus the default validator. Encapsulating this logic also makes it easier to mock/test.